docs/src/pages/docs/crocks/Result.md

```diff
@@ -0,0 +1,1422 @@
+//=> false
+```
+
+#### concat
+
+```haskell
+Semigroup s => Result e s ~> Result e s -> Result e s
+```
+
+When an underlying value of a given `Result` is fixed to a `Semigroup`, `concat`
+can be used to concat another `Result` instance with an underlying `Semigroup`
+of the same type. Expecting a `Result` wrapping a `Semigroup` of the same type,
+`concat` will give back a new `Result` instance wrapping the result of combining
+the two underlying `Semigroup`s. When called on a [`Err`](#err) instance,
+`concat` will return an [`Err`](#err) with the value of the first [`Err`](#err)
+in the chain.
+
+```javascript
+import Result from 'crocks/Result'
+
+import concat from 'crocks/pointfree/concat'
+
+const { Ok, Err } = Result
+
+Ok([ 1, 2, 3 ])
+  .concat(Ok([ 4, 5, 6 ]))
+//=> [ 1, 2, 3, 4, 5, 6 ]
+
+Ok([ 1, 2, 3 ])
+  .concat(Err([ 4, 5, 6 ]))
+//=> Err [ 4, 5, 6 ]
+
+concat(Ok('Result'), Err('Error'))
+//=> Err "Error"
+
+concat(Err('Error'), Ok('Result'))
+//=> Err "Error"
+
+concat(Err('Error 1'), Err('Error 2'))
+//=> Err "Error 1"
+```
+
+#### map
+
+```haskell
+Result e a ~> (a -> b) -> Result e b
+```
+
+Used to apply transformations to values in the safety of a `Result`, `map` takes
+a function that it will lift into the context of the `Result` and apply to it
+the wrapped value. When run on an [`Ok`](#ok) instance, `map` will apply the
+wrapped value to the provided function and return the result in a new [`Ok`](#ok)
+instance. When run on an ['Err'](#err) `map` with return the error value in a new
+[`Err`](#err) instance.
+
+```javascript
+import Result from 'crocks/Result'
+
+import map from 'crocks/pointfree/map'
+import merge from 'crocks/pointfree/merge'
+import assign from 'crocks/helpers/assign'
+import objOf from 'crocks/helpers/objOf'
+import fanout from 'crocks/Pair/fanout'
+import isNumber from 'crocks/predicates/isNumber'
+import ifElse from 'crocks/logic/ifElse'
+import compose from 'crocks/core/compose'
+
+const { Ok, Err } = Result
+
+// buildError :: String -> String
+const buildError = x => Err(`${x} is not a valid number`)
+
+// prod :: Number -> Number -> Number
+const prod =
+  x => y => x * y
+
+// fromNumber :: a -> Result
+const fromNumber =
+  ifElse(isNumber, Ok, buildError)
+
+// doube :: a -> Result String Number
+const double =
+  compose(
+    map(prod(2)),
+    fromNumber
+  )
+
+double(21)
+//=> Ok 42
+
+double('down')
+//=> Err "down is not a valid number"
+
+// getParity :: Number -> Object
+const getParity = compose(
+  objOf('parity'),
+  n => n % 2 === 0 ? 'Even' : 'Odd'
+)
+
+// getInfo :: a -> Object
+const getInfo = compose(
+  map(merge(assign)),
+  map(fanout(objOf('value'), getParity)),
+  fromNumber
+)
+
+getInfo(5324)
+//=> Ok { parity: "Even", value: 5324 }
+
+getInfo('down')
+//=> Err "down is not a valid number"
+```
+
+#### alt
+
+```haskell
+Result e a ~> Result e a -> Result e a
+```
+
+Providing a means for a fallback or alternative value, `alt` combines two
```

Comment:
One thing that is missed with `alt` is that fact that `Err` of `Semigroup` will accumulate like `ap` does.
Should be mentioned in the text and shown in the examples.

Reply:
I thought it did but when i tested it that didn't seem to happen so i was confused. More than likely i made a mistake so will go back and try again

Reply:
I see where i made the mix-up, i knew there was this behaviour but i thought it was on `concat`. Will update `alt` and add the examples